### PR TITLE
add Advanced search and Particular search

### PR DESCRIPTION
**Pull request by Rashmil-1999**
This commit is a mixture of Nirav's work and mine.
I made some minor tweaks in Nirav's work and added some new things which enables the user to make queries on any model about any particular thing say only inhouse projects so the query will look like this 
`GET api/projects/?is_inhouse=false&approved=true`
also for any random search 
`GET api/search/?generic=<search anything and it will return response from contributer and  project model based on name>`
Other Issues also fixed. 
Work left is the addition of permission classes and will be done today.
so basically the `Advanced Search` is implemented.

### CODE REVIEW

File: BEProjectsApp/models.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 from django.contrib.auth.models import User
 
 DOMAIN_CHOICES = [
-    ("1", ("Data Mining & Analytics")),
+    ("Data Mining & Analytics", ("Data Mining & Analytics")),
```
Comment (pujanm) [inline BEProjectsApp/models.py]: Why is this different then others?

**Comment by pujanm** [comment]
@Rashmil-1999 your build is failing, please run `black` command. Then I can merge your PR.

File: BEProjectsApp/models.py (hunk 1 above)
Comment (Rashmil-1999) [reply BEProjectsApp/models.py]: due to the previous one the data which was stored in database was numbers as string now it will be actual data

Comment (Rashmil-1999) [reply BEProjectsApp/models.py]: latest commit is perfect
